# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: D:\projects\foreign\apache\jmeter\trunk
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: build.properties
--- build.properties Base (BASE)
+++ build.properties Locally Modified (Based On LOCAL)
@@ -400,7 +400,7 @@
 rat-tasks.md5                = 96b699581b4475ed5756a0c24af745e8
 
 # Optional for use by JaCoCo
-jacocoant.version            = 0.7.7.201606060606
+jacocoant.version            = 0.7.8
 jacocoant.jar                = org.jacoco.ant-${jacocoant.version}-nodeps.jar
 jacocoant.loc                = ${maven2.repo}/org/jacoco/org.jacoco.ant/${jacocoant.version}
-jacocoant.md5                = fc093f67abf5bafb782d2eafa2f5815e
+jacocoant.md5                = e308c0fad91f19e1f485efc15c4395e6
Index: build.xml
--- build.xml Base (BASE)
+++ build.xml Locally Modified (Based On LOCAL)
@@ -18,7 +18,6 @@
 <project name="JMeter" default="install" basedir="."
     xmlns:rat="antlib:org.apache.rat.anttasks"
     xmlns:jacoco="antlib:org.jacoco.ant"
-    xmlns:if="ant:if"
     xmlns="antlib:org.apache.tools.ant" >
   <description>
 
@@ -880,14 +879,14 @@
   description="Compile everything."/>
 
   <target name="run_gui" depends="package" description="Run the JMeter GUI off the jar files">
-    <condition property="jacoco_params" value="${jacocoagent}run_gui:java(server) org.apache.jmeter.NewDriver" else="-Djacoco_dummy=not_empty">
+    <condition property="jacoco_params" value="${jacocoagent}" else="-Ddummy">
       <isset property="jacocoagent"/>
     </condition>
     <java classname="org.apache.jmeter.NewDriver" fork="true">
       <classpath>
         <pathelement location="${dest.jar.jmeter}/ApacheJMeter.jar"/>
       </classpath>
-      <jvmarg value="${jacoco_params}"/>
+      <jvmarg value="${jacoco_params}run_gui:java org.apache.jmeter.NewDriver"/>
       <sysproperty key="jmeter.home" value="${basedir}"/>
     </java>
   </target>
@@ -2436,7 +2435,7 @@
     <condition property="rmi_hostname_option" value="-Djava.rmi.server.hostname=localhost">
       <isset property="rmi_force_localhost"/>
     </condition>
-    <condition property="jacoco_params" value="${jacocoagent}batchtest:daemon:java(server) -Djava.awt.headless=true org.apache.jmeter.NewDriver -j ${batchtestserver.out}/${batchtestserver.log} -Dserver_port=${rmi_port}" else="-Djacoco_dummy=not_empty">
+    <condition property="jacoco_params" value="${jacocoagent}" else="-Djacoco_dummy">
       <isset property="jacocoagent"/>
     </condition>
     <tstamp>
@@ -2451,7 +2450,7 @@
             <fileset dir="${dest.jar}" includes="*.jar"/>
             <path refid="classpath"/>
           </classpath>
-          <jvmarg value="${jacoco_params}"/>
+          <jvmarg value="${jacoco_params}batchtest:daemon:java(server) -Djava.awt.headless=true org.apache.jmeter.NewDriver -j ${batchtestserver.out}/${batchtestserver.log} -Dserver_port=${rmi_port}"/>
           <sysproperty key="java.awt.headless" value="true"/>
           <!-- Bug 59723 -->
           <!-- quieten the logging; this has to be done first -->
@@ -2548,7 +2547,7 @@
 
     <deleteworkfiles/>
 
-    <condition property="jacoco_params" value="${jacocoagent}batchtest:java(${taskname}) -Djava.awt.headless=false org.apache.jmeter.NewDriver -ttestfiles/${batchtest.name}.jmx ${remote}" else="-Djacoco_dummy=not_empty">
+    <condition property="jacoco_params" value="${jacocoagent}" else="-Djacoco_dummy=">
       <isset property="jacocoagent"/>
     </condition>
 
@@ -2558,7 +2557,7 @@
         <fileset dir="${dest.jar}" includes="*.jar"/>
         <path refid="classpath"/>
       </classpath>
-      <jvmarg value="${jacoco_params}"/>
+      <jvmarg value="${jacoco_params}batchtest:java(${taskname}) -Djava.awt.headless=false org.apache.jmeter.NewDriver -ttestfiles/${batchtest.name}.jmx ${remote}" />
       <!-- Detect if non-GUI runs OK headless by forcing it to try using non-headless mode -->
       <sysproperty key="user.language" value="en"/>
       <sysproperty key="user.region" value="US"/>
@@ -2749,7 +2748,7 @@
     <delete quiet="true">
       <fileset dir="${basedir}/bin/testfiles" includes="*.jmx.out"/>
     </delete>
-    <condition property="jacoco_params" value="${jacocoagent}_test:java -Djava.awt.headless=${test.headless} org.apache.jorphan.test.AllTests" else="-Djacoco_dummy=not_empty">
+    <condition property="jacoco_params" value="${jacocoagent}" else="-Djacoco_dummy=">
       <isset property="jacocoagent"/>
     </condition>
     <!-- fork="yes" is required or dir attribute is ignored -->
@@ -2762,7 +2761,7 @@
         <path refid="classpath"/>
       </classpath>
       <jvmarg value="-server"/>
-      <jvmarg value="${jacoco_params}"/>
+      <jvmarg value="${jacoco_params}_test:java -Djava.awt.headless=${test.headless} org.apache.jorphan.test.AllTests" />
       <jvmarg value="-Dfile.encoding=${test.encoding}"/>
       <sysproperty key="java.awt.headless" value="${test.headless}"/>
       <sysproperty key="testsaveservice.saveout" value="${testsaveservice.saveout}" />
@@ -2783,7 +2782,7 @@
     <property name="test.format" value="plain"/>
     <property name="test.case" value="org.apache.jorphan.test.AllTests"/>
     <property name="test.showoutput" value="false"/>
-    <condition property="jacoco_params" value="${jacocoagent}run_gui:junit -Djava.awt.headless=${java.awt.headless} org.apache.jmeter.NewDriver" else="-Djacoco_dummy=not_empty">
+    <condition property="jacoco_params" value="${jacocoagent}junit -Djava.awt.headless=${java.awt.headless}" else="-Djacoco_dummy">
       <isset property="jacocoagent"/>
     </condition>
     <junit fork="true"
@@ -2792,7 +2791,7 @@
            printsummary="on">
       <formatter type="${test.format}" usefile="yes"/>
       <formatter type="xml"/>
-      <jvmarg value="${jacoco_params}"/>
+      <jvmarg value="${jacoco_params}junit:junit testname=${test.case}" />
       <jvmarg value="-Dfile.encoding=${test.encoding}"/>
       <!-- Allow headless to be passed in -->
       <jvmarg value="-Djava.awt.headless=${java.awt.headless}"/>
@@ -3142,6 +3141,10 @@
      <taskdef uri="antlib:org.jacoco.ant" resource="org/jacoco/ant/antlib.xml">
         <classpath path="${lib.coverage}/${jacocoant.jar}"/>
      </taskdef>
+     <!-- Please note that the resulting property has a blank session Id parameter at the end,
+     ! wich will be used to indicate / track individual executions in java & junit tasks. 
+     ! The sessions will be reported in reports/coverage/jacoco-sessions.html
+     !-->
      <jacoco:agent property="jacocoagent" 
                    destfile="${build.dir}/jacoco.exec"
                    includes="org.apache.jmeter.*:org.apache.jorphan.*:org.apache.commons.cli.avalon.*:org.sl4j.impl.StaticLoggerBinder"
@@ -3154,7 +3157,7 @@
         <executiondata>
             <file file="${build.dir}/jacoco.exec"/>
         </executiondata>
-        <structure name="JMeter ${jmeter.version} ${implementation.version}">
+        <structure name="JMeter ${jmeter.version} Implementation ${implementation.version} SVN Rev ${svn.revision}">
             <classfiles>
                 <fileset dir="${build.components}"/>
                 <fileset dir="${build.core}"/>
